README.md
# flutter_iconly

Flutter package for [Iconly 2.3 Essential Icons](https://www.figma.com/community/file/876509330914541878). Flutter iconly is package of more than 300 icons in 3 styles(Light, Bold, Broken).

You can see list name of Flutter Iconly in [Flutter Iconly Web](https://m-noer.github.io/flutter_iconly_web/)

## Screenshot   
![Flutter Iconly Web](assets/flutter_iconly_web.png)

## Installation
In the dependencies: section of your pubspec.yaml, add the following line:
```yaml
flutter_iconly: <latest_verion>
```

## Usage
```dart
import 'package:flutter/material.dart';
import 'package:flutter_iconly/flutter_iconly.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Iconly Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: SafeArea(
          child: Center(
            child: Column(
              children: [
                /// Icon Style Bold
                Icon(IconlyBold.user2),

                /// Icon Style Light
                Icon(IconlyLight.activity),

                /// Icon Style Broken
                Icon(IconlyBroken.user2),

                Icon(Icons.ac_unit)
              ],
            ),
          ),
        ),
      ),
    );
  }
}

```



CHANGELOG.md
## 1.0.2

* Update readme

## 1.0.1
* Fix bug icon not shown
## 1.0.0

* Initial release


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_iconly/flutter_iconly.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Iconly Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: SafeArea(
          child: Center(
            child: Column(
              children: [
                /// Icon Style Bold
                Icon(IconlyBold.user2),

                /// Icon Style Light
                Icon(IconlyLight.activity),

                /// Icon Style Broken
                Icon(IconlyBroken.user2),

                Icon(Icons.ac_unit)
              ],
            ),
          ),
        ),
      ),
    );
  }
}


lib/src/iconly_bold.dart
// GENERATED CODE - DO NOT MODIFY BY HAND

import 'package:flutter/material.dart';

/// [IconlyBoldData] for style Bold Icons
///
class IconlyBoldData extends IconData {
  const IconlyBoldData(int codePoint)
      : super(
          codePoint,
          fontFamily: 'IconlyBold',
          fontPackage: 'flutter_iconly',
        );
}

/// [IconlyBold] for style Bold Icons
///
/// List Iconly Bold
///
class IconlyBold {
  IconlyBold._();

  static const IconData user2 = IconlyBoldData(0xe000);
  static const IconData user3 = IconlyBoldData(0xe001);
  static const IconData activity = IconlyBoldData(0xe002);
  static const IconData addUser = IconlyBoldData(0xe003);
  static const IconData arrowDown2 = IconlyBoldData(0xe004);
  static const IconData arrowDown3 = IconlyBoldData(0xe005);
  static const IconData arrowDownCircle = IconlyBoldData(0xe006);
  static const IconData arrowDownSquare = IconlyBoldData(0xe007);
  static const IconData arrowDown = IconlyBoldData(0xe008);
  static const IconData arrowLeft2 = IconlyBoldData(0xe009);
  static const IconData arrowLeft3 = IconlyBoldData(0xe00a);
  static const IconData arrowLeftCircle = IconlyBoldData(0xe00b);
  static const IconData arrowLeftSquare = IconlyBoldData(0xe00c);
  static const IconData arrowLeft = IconlyBoldData(0xe00d);
  static const IconData arrowRight2 = IconlyBoldData(0xe00e);
  static const IconData arrowRight3 = IconlyBoldData(0xe00f);
  static const IconData arrowRightCircle = IconlyBoldData(0xe010);
  static const IconData arrowRightSquare = IconlyBoldData(0xe011);
  static const IconData arrowRight = IconlyBoldData(0xe012);
  static const IconData arrowUp2 = IconlyBoldData(0xe013);
  static const IconData arrowUp3 = IconlyBoldData(0xe014);
  static const IconData arrowUpCircle = IconlyBoldData(0xe015);
  static const IconData arrowUpSquare = IconlyBoldData(0xe016);
  static const IconData arrowUp = IconlyBoldData(0xe017);
  static const IconData bag2 = IconlyBoldData(0xe018);
  static const IconData bag = IconlyBoldData(0xe019);
  static const IconData bookmark = IconlyBoldData(0xe01a);
  static const IconData buy = IconlyBoldData(0xe01b);
  static const IconData calendar = IconlyBoldData(0xe01c);
  static const IconData callMissed = IconlyBoldData(0xe01d);
  static const IconData callSilent = IconlyBoldData(0xe01e);
  static const IconData call = IconlyBoldData(0xe01f);
  static const IconData calling = IconlyBoldData(0xe020);
  static const IconData camera = IconlyBoldData(0xe021);
  static const IconData category = IconlyBoldData(0xe022);
  static const IconData chart = IconlyBoldData(0xe023);
  static const IconData chat = IconlyBoldData(0xe024);
  static const IconData closeSquare = IconlyBoldData(0xe025);
  static const IconData danger = IconlyBoldData(0xe026);
  static const IconData delete = IconlyBoldData(0xe027);
  static const IconData discount = IconlyBoldData(0xe028);
  static const IconData discovery = IconlyBoldData(0xe029);
  static const IconData document = IconlyBoldData(0xe02a);
  static const IconData download = IconlyBoldData(0xe02b);
  static const IconData editSquare = IconlyBoldData(0xe02c);
  static const IconData edit = IconlyBoldData(0xe02d);
  static const IconData filter2 = IconlyBoldData(0xe02e);
  static const IconData filter = IconlyBoldData(0xe02f);
  static const IconData folder = IconlyBoldData(0xe030);
  static const IconData game = IconlyBoldData(0xe031);
  static const IconData graph = IconlyBoldData(0xe032);
  static const IconData heart = IconlyBoldData(0xe033);
  static const IconData hide = IconlyBoldData(0xe034);
  static const IconData home = IconlyBoldData(0xe035);
  static const IconData image2 = IconlyBoldData(0xe036);
  static const IconData image = IconlyBoldData(0xe037);
  static const IconData infoCircle = IconlyBoldData(0xe038);
  static const IconData infoSquare = IconlyBoldData(0xe039);
  static const IconData location = IconlyBoldData(0xe03a);
  static const IconData lock = IconlyBoldData(0xe03b);
  static const IconData login = IconlyBoldData(0xe03c);
  static const IconData logout = IconlyBoldData(0xe03d);
  static const IconData message = IconlyBoldData(0xe03e);
  static const IconData moreCircle = IconlyBoldData(0xe03f);
  static const IconData moreSquare = IconlyBoldData(0xe040);
  static const IconData notification = IconlyBoldData(0xe041);
  static const IconData paperDownload = IconlyBoldData(0xe042);
  static const IconData paperFail = IconlyBoldData(0xe043);
  static const IconData paperNegative = IconlyBoldData(0xe044);
  static const IconData paperPlus = IconlyBoldData(0xe045);
  static const IconData paperUpload = IconlyBoldData(0xe046);
  static const IconData paper = IconlyBoldData(0xe047);
  static const IconData password = IconlyBoldData(0xe048);
  static const IconData play = IconlyBoldData(0xe049);
  static const IconData plus = IconlyBoldData(0xe04a);
  static const IconData profile = IconlyBoldData(0xe04b);
  static const IconData scan = IconlyBoldData(0xe04c);
  static const IconData search = IconlyBoldData(0xe04d);
  static const IconData send = IconlyBoldData(0xe04e);
  static const IconData setting = IconlyBoldData(0xe04f);
  static const IconData shieldDone = IconlyBoldData(0xe050);
  static const IconData shieldFail = IconlyBoldData(0xe051);
  static const IconData show = IconlyBoldData(0xe052);
  static const IconData star = IconlyBoldData(0xe053);
  static const IconData swap = IconlyBoldData(0xe054);
  static const IconData tickSquare = IconlyBoldData(0xe055);
  static const IconData ticketStar = IconlyBoldData(0xe056);
  static const IconData ticket = IconlyBoldData(0xe057);
  static const IconData timeCircle = IconlyBoldData(0xe058);
  static const IconData timeSquare = IconlyBoldData(0xe059);
  static const IconData unlock = IconlyBoldData(0xe05a);
  static const IconData upload = IconlyBoldData(0xe05b);
  static const IconData video = IconlyBoldData(0xe05c);
  static const IconData voice2 = IconlyBoldData(0xe05d);
  static const IconData voice = IconlyBoldData(0xe05e);
  static const IconData volumeDown = IconlyBoldData(0xe05f);
  static const IconData volumeOff = IconlyBoldData(0xe060);
  static const IconData volumeUp = IconlyBoldData(0xe061);
  static const IconData wallet = IconlyBoldData(0xe062);
  static const IconData work = IconlyBoldData(0xe063);
}


lib/src/iconly_light.dart
// GENERATED CODE - DO NOT MODIFY BY HAND

import 'package:flutter/material.dart';

/// [IconlyLightData] for style Light
///
class IconlyLightData extends IconData {
  const IconlyLightData(int codePoint)
      : super(
          codePoint,
          fontFamily: 'IconlyLight',
          fontPackage: 'flutter_iconly',
        );
}

/// [IconlyLight] for style
///
/// List Iconly Light
///

class IconlyLight {
  IconlyLight._();

  static const IconData user2 = IconlyLightData(0xe000);
  static const IconData user3 = IconlyLightData(0xe001);
  static const IconData activity = IconlyLightData(0xe002);
  static const IconData addUser = IconlyLightData(0xe003);
  static const IconData arrowDown2 = IconlyLightData(0xe004);
  static const IconData arrowDown3 = IconlyLightData(0xe005);
  static const IconData arrowDownCircle = IconlyLightData(0xe006);
  static const IconData arrowDownSquare = IconlyLightData(0xe007);
  static const IconData arrowDown = IconlyLightData(0xe008);
  static const IconData arrowLeft2 = IconlyLightData(0xe009);
  static const IconData arrowLeft3 = IconlyLightData(0xe00a);
  static const IconData arrowLeftCircle = IconlyLightData(0xe00b);
  static const IconData arrowLeftSquare = IconlyLightData(0xe00c);
  static const IconData arrowLeft = IconlyLightData(0xe00d);
  static const IconData arrowRight2 = IconlyLightData(0xe00e);
  static const IconData arrowRight3 = IconlyLightData(0xe00f);
  static const IconData arrowRightCircle = IconlyLightData(0xe010);
  static const IconData arrowRightSquare = IconlyLightData(0xe011);
  static const IconData arrowRight = IconlyLightData(0xe012);
  static const IconData arrowUp2 = IconlyLightData(0xe013);
  static const IconData arrowUp3 = IconlyLightData(0xe014);
  static const IconData arrowUpCircle = IconlyLightData(0xe015);
  static const IconData arrowUpSquare = IconlyLightData(0xe016);
  static const IconData arrowUp = IconlyLightData(0xe017);
  static const IconData bag2 = IconlyLightData(0xe018);
  static const IconData bag = IconlyLightData(0xe019);
  static const IconData bookmark = IconlyLightData(0xe01a);
  static const IconData buy = IconlyLightData(0xe01b);
  static const IconData calendar = IconlyLightData(0xe01c);
  static const IconData callMissed = IconlyLightData(0xe01d);
  static const IconData callSilent = IconlyLightData(0xe01e);
  static const IconData call = IconlyLightData(0xe01f);
  static const IconData calling = IconlyLightData(0xe020);
  static const IconData camera = IconlyLightData(0xe021);
  static const IconData category = IconlyLightData(0xe022);
  static const IconData chart = IconlyLightData(0xe023);
  static const IconData chat = IconlyLightData(0xe024);
  static const IconData closeSquare = IconlyLightData(0xe025);
  static const IconData dangerCircle = IconlyLightData(0xe026);
  static const IconData dangerTriangle = IconlyLightData(0xe027);
  static const IconData delete = IconlyLightData(0xe028);
  static const IconData discount = IconlyLightData(0xe029);
  static const IconData discovery = IconlyLightData(0xe02a);
  static const IconData document = IconlyLightData(0xe02b);
  static const IconData download = IconlyLightData(0xe02c);
  static const IconData editSquare = IconlyLightData(0xe02d);
  static const IconData edit = IconlyLightData(0xe02e);
  static const IconData filter2 = IconlyLightData(0xe02f);
  static const IconData filter = IconlyLightData(0xe030);
  static const IconData folder = IconlyLightData(0xe031);
  static const IconData game = IconlyLightData(0xe032);
  static const IconData graph = IconlyLightData(0xe033);
  static const IconData heart = IconlyLightData(0xe034);
  static const IconData hide = IconlyLightData(0xe035);
  static const IconData home = IconlyLightData(0xe036);
  static const IconData image2 = IconlyLightData(0xe037);
  static const IconData image = IconlyLightData(0xe038);
  static const IconData infoSquare = IconlyLightData(0xe039);
  static const IconData location = IconlyLightData(0xe03a);
  static const IconData lock = IconlyLightData(0xe03b);
  static const IconData login = IconlyLightData(0xe03c);
  static const IconData logout = IconlyLightData(0xe03d);
  static const IconData message = IconlyLightData(0xe03e);
  static const IconData moreCircle = IconlyLightData(0xe03f);
  static const IconData moreSquare = IconlyLightData(0xe040);
  static const IconData notification = IconlyLightData(0xe041);
  static const IconData paperDownload = IconlyLightData(0xe042);
  static const IconData paperFail = IconlyLightData(0xe043);
  static const IconData paperNegative = IconlyLightData(0xe044);
  static const IconData paperPlus = IconlyLightData(0xe045);
  static const IconData paperUpload = IconlyLightData(0xe046);
  static const IconData paper = IconlyLightData(0xe047);
  static const IconData password = IconlyLightData(0xe048);
  static const IconData play = IconlyLightData(0xe049);
  static const IconData plus = IconlyLightData(0xe04a);
  static const IconData profile = IconlyLightData(0xe04b);
  static const IconData scan = IconlyLightData(0xe04c);
  static const IconData search = IconlyLightData(0xe04d);
  static const IconData send = IconlyLightData(0xe04e);
  static const IconData setting = IconlyLightData(0xe04f);
  static const IconData shieldDone = IconlyLightData(0xe050);
  static const IconData shieldFail = IconlyLightData(0xe051);
  static const IconData show = IconlyLightData(0xe052);
  static const IconData star = IconlyLightData(0xe053);
  static const IconData swap = IconlyLightData(0xe054);
  static const IconData tickSquare = IconlyLightData(0xe055);
  static const IconData ticketStar = IconlyLightData(0xe056);
  static const IconData ticket = IconlyLightData(0xe057);
  static const IconData timeCircle = IconlyLightData(0xe058);
  static const IconData timeSquare = IconlyLightData(0xe059);
  static const IconData unlock = IconlyLightData(0xe05a);
  static const IconData upload = IconlyLightData(0xe05b);
  static const IconData video = IconlyLightData(0xe05c);
  static const IconData voice2 = IconlyLightData(0xe05d);
  static const IconData voice = IconlyLightData(0xe05e);
  static const IconData volumeDown = IconlyLightData(0xe05f);
  static const IconData volumeOff = IconlyLightData(0xe060);
  static const IconData volumeUp = IconlyLightData(0xe061);
  static const IconData wallet = IconlyLightData(0xe062);
  static const IconData work = IconlyLightData(0xe063);
}


lib/src/iconly_broken.dart
// GENERATED CODE - DO NOT MODIFY BY HAND

import 'package:flutter/material.dart';

/// [IconlyBrokenData] for style Broken Icons
///
class IconlyBrokenData extends IconData {
  const IconlyBrokenData(int codePoint)
      : super(
          codePoint,
          fontFamily: 'IconlyBroken',
          fontPackage: 'flutter_iconly',
        );
}

/// [IconlyBroken] for style Broken Icons
///
/// List Iconly Broken
///
class IconlyBroken {
  IconlyBroken._();

  static const IconData user2 = IconlyBrokenData(0xe000);
  static const IconData user3 = IconlyBrokenData(0xe001);
  static const IconData activity = IconlyBrokenData(0xe002);
  static const IconData addUser = IconlyBrokenData(0xe003);
  static const IconData arrowDown2 = IconlyBrokenData(0xe004);
  static const IconData arrowDown3 = IconlyBrokenData(0xe005);
  static const IconData arrowDownCircle = IconlyBrokenData(0xe006);
  static const IconData arrowDownSquare = IconlyBrokenData(0xe007);
  static const IconData arrowDown = IconlyBrokenData(0xe008);
  static const IconData arrowLeft2 = IconlyBrokenData(0xe009);
  static const IconData arrowLeft3 = IconlyBrokenData(0xe00a);
  static const IconData arrowLeftCircle = IconlyBrokenData(0xe00b);
  static const IconData arrowLeftSquare = IconlyBrokenData(0xe00c);
  static const IconData arrowLeft = IconlyBrokenData(0xe00d);
  static const IconData arrowRight2 = IconlyBrokenData(0xe00e);
  static const IconData arrowRight3 = IconlyBrokenData(0xe00f);
  static const IconData arrowRightCircle = IconlyBrokenData(0xe010);
  static const IconData arrowRightSquare = IconlyBrokenData(0xe011);
  static const IconData arrowRight = IconlyBrokenData(0xe012);
  static const IconData arrowUp2 = IconlyBrokenData(0xe013);
  static const IconData arrowUp3 = IconlyBrokenData(0xe014);
  static const IconData arrowUpCircle = IconlyBrokenData(0xe015);
  static const IconData arrowUpSquare = IconlyBrokenData(0xe016);
  static const IconData arrowUp = IconlyBrokenData(0xe017);
  static const IconData bag2 = IconlyBrokenData(0xe018);
  static const IconData bag = IconlyBrokenData(0xe019);
  static const IconData bookmark = IconlyBrokenData(0xe01a);
  static const IconData buy = IconlyBrokenData(0xe01b);
  static const IconData calendar = IconlyBrokenData(0xe01c);
  static const IconData callMissed = IconlyBrokenData(0xe01d);
  static const IconData callSilent = IconlyBrokenData(0xe01e);
  static const IconData call = IconlyBrokenData(0xe01f);
  static const IconData calling = IconlyBrokenData(0xe020);
  static const IconData camera = IconlyBrokenData(0xe021);
  static const IconData category = IconlyBrokenData(0xe022);
  static const IconData chart = IconlyBrokenData(0xe023);
  static const IconData chat = IconlyBrokenData(0xe024);
  static const IconData closeSquare = IconlyBrokenData(0xe025);
  static const IconData danger = IconlyBrokenData(0xe026);
  static const IconData delete = IconlyBrokenData(0xe027);
  static const IconData discount = IconlyBrokenData(0xe028);
  static const IconData discovery = IconlyBrokenData(0xe029);
  static const IconData document = IconlyBrokenData(0xe02a);
  static const IconData download = IconlyBrokenData(0xe02b);
  static const IconData editSquare = IconlyBrokenData(0xe02c);
  static const IconData edit = IconlyBrokenData(0xe02d);
  static const IconData filter2 = IconlyBrokenData(0xe02e);
  static const IconData filter = IconlyBrokenData(0xe02f);
  static const IconData folder = IconlyBrokenData(0xe030);
  static const IconData game = IconlyBrokenData(0xe031);
  static const IconData graph = IconlyBrokenData(0xe032);
  static const IconData heart = IconlyBrokenData(0xe033);
  static const IconData hide = IconlyBrokenData(0xe034);
  static const IconData home = IconlyBrokenData(0xe035);
  static const IconData image2 = IconlyBrokenData(0xe036);
  static const IconData image = IconlyBrokenData(0xe037);
  static const IconData infoCircle = IconlyBrokenData(0xe038);
  static const IconData infoSquare = IconlyBrokenData(0xe039);
  static const IconData location = IconlyBrokenData(0xe03a);
  static const IconData lock = IconlyBrokenData(0xe03b);
  static const IconData login = IconlyBrokenData(0xe03c);
  static const IconData logout = IconlyBrokenData(0xe03d);
  static const IconData message = IconlyBrokenData(0xe03e);
  static const IconData moreCircle = IconlyBrokenData(0xe03f);
  static const IconData moreSquare = IconlyBrokenData(0xe040);
  static const IconData notification = IconlyBrokenData(0xe041);
  static const IconData paperDownload = IconlyBrokenData(0xe042);
  static const IconData paperFail = IconlyBrokenData(0xe043);
  static const IconData paperNegative = IconlyBrokenData(0xe044);
  static const IconData paperPlus = IconlyBrokenData(0xe045);
  static const IconData paperUpload = IconlyBrokenData(0xe046);
  static const IconData paper = IconlyBrokenData(0xe047);
  static const IconData password = IconlyBrokenData(0xe048);
  static const IconData play = IconlyBrokenData(0xe049);
  static const IconData plus = IconlyBrokenData(0xe04a);
  static const IconData profile = IconlyBrokenData(0xe04b);
  static const IconData scan = IconlyBrokenData(0xe04c);
  static const IconData search = IconlyBrokenData(0xe04d);
  static const IconData send = IconlyBrokenData(0xe04e);
  static const IconData setting = IconlyBrokenData(0xe04f);
  static const IconData shieldDone = IconlyBrokenData(0xe050);
  static const IconData shieldFail = IconlyBrokenData(0xe051);
  static const IconData show = IconlyBrokenData(0xe052);
  static const IconData star = IconlyBrokenData(0xe053);
  static const IconData swap = IconlyBrokenData(0xe054);
  static const IconData tickSquare = IconlyBrokenData(0xe055);
  static const IconData ticketStar = IconlyBrokenData(0xe056);
  static const IconData ticket = IconlyBrokenData(0xe057);
  static const IconData timeCircle = IconlyBrokenData(0xe058);
  static const IconData timeSquare = IconlyBrokenData(0xe059);
  static const IconData unlock = IconlyBrokenData(0xe05a);
  static const IconData upload = IconlyBrokenData(0xe05b);
  static const IconData video = IconlyBrokenData(0xe05c);
  static const IconData voice2 = IconlyBrokenData(0xe05d);
  static const IconData voice = IconlyBrokenData(0xe05e);
  static const IconData volumeDown = IconlyBrokenData(0xe05f);
  static const IconData volumeOff = IconlyBrokenData(0xe060);
  static const IconData volumeUp = IconlyBrokenData(0xe061);
  static const IconData wallet = IconlyBrokenData(0xe062);
  static const IconData work = IconlyBrokenData(0xe063);
}
